data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/WorkerCommonUtils.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.opensearch.worker;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSetManager;
+import org.opensearch.dataprepper.model.source.coordinator.SourceCoordinator;
+import org.opensearch.dataprepper.model.source.coordinator.SourcePartition;
+import org.opensearch.dataprepper.plugins.source.opensearch.OpenSearchIndexProgressState;
+import org.opensearch.dataprepper.plugins.source.opensearch.OpenSearchSourceConfiguration;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class WorkerCommonUtils {
+
+    static final int ACKNOWLEDGEMENT_SET_TIMEOUT_SECONDS = Integer.MAX_VALUE;
+
+    static Pair<AcknowledgementSet, CompletableFuture<Boolean>> createAcknowledgmentSet(final AcknowledgementSetManager acknowledgementSetManager,
+                                                                                               final OpenSearchSourceConfiguration openSearchSourceConfiguration,
+                                                                                               final SourceCoordinator<OpenSearchIndexProgressState> sourceCoordinator,
+                                                                                               final SourcePartition<OpenSearchIndexProgressState> indexPartition) {
+        AcknowledgementSet acknowledgementSet = null;
+        CompletableFuture<Boolean> completableFuture = new CompletableFuture<>();
+
+        if (openSearchSourceConfiguration.isAcknowledgmentsEnabled()) {
+            acknowledgementSet = acknowledgementSetManager.create((result) -> {
+                if (result == true) {
+                    sourceCoordinator.closePartition(
+                            indexPartition.getPartitionKey(),
+                            openSearchSourceConfiguration.getSchedulingParameterConfiguration().getRate(),
+                            openSearchSourceConfiguration.getSchedulingParameterConfiguration().getJobCount());
+                }
```

Comment:
Do you want to handle `result==false` case? We don't have any code doing negative acks, so it is not mandatory now. But may be used in future.

Reply:
Thanks for the callout. Just curious how are we planning on handling negative acks? Just retrying the entire object, index, etc?